dipy/reconst/fwdti.py

```diff
@@ -0,0 +1,876 @@
+    direction on the sphere for which a signal is to be predicted, $b$ is the b
+    value provided in the GradientTable input for that direction, $Q$ is the
+    quadratic form of the tensor determined by the input parameters, $f$ is the
+    free water diffusion compartment, $D_{iso}$ is the free water diffusivity
+    which is equal to $3 * 10^{-3} mm^{2}s^{-1} [1]_.
+
+    References
+    ----------
+    .. [1] Hoy, A.R., Koay, C.G., Kecskemeti, S.R., Alexander, A.L., 2014.
+           Optimization of a free water elimination two-compartmental model
+           for diffusion tensor imaging. NeuroImage 103, 323-333.
+           doi: 10.1016/j.neuroimage.2014.09.053
+    """
+    evals = params[..., :3]
+    evecs = params[..., 3:-1].reshape(params.shape[:-1] + (3, 3))
+    f = params[..., 12]
+    qform = vec_val_vect(evecs, evals)
+    lower_dt = lower_triangular(qform, S0)
+    lower_diso = lower_dt.copy()
+    lower_diso[..., 0] = lower_diso[..., 2] = lower_diso[..., 5] = Diso
+    lower_diso[..., 1] = lower_diso[..., 3] = lower_diso[..., 4] = 0
+    D = design_matrix(gtab)
+
+    pred_sig = np.zeros(f.shape + (gtab.bvals.shape[0],))
+    mask = _positive_evals(evals[..., 0], evals[..., 1], evals[..., 2])
+    index = ndindex(f.shape)
+    for v in index:
+        if mask[v]:
+            pred_sig[v] = (1 - f[v]) * np.exp(np.dot(lower_dt[v], D.T)) + \
+                          f[v] * np.exp(np.dot(lower_diso[v], D.T))
+
+    return pred_sig
+
+
+class FreeWaterTensorModel(ReconstModel):
+    """ Class for the Free Water Elimination Diffusion Tensor Model """
+    def __init__(self, gtab, fit_method="NLS", *args, **kwargs):
+        """ Free Water Diffusion Tensor Model [1]_.
+
+        Parameters
+        ----------
+        gtab : GradientTable class instance
+        fit_method : str or callable
+            str can be one of the following:
+
+            'WLS' for weighted linear least square fit according to [1]_
+                :func:`fwdti.wls_fit_tensor`
+            'NLS' for non-linear least square fit according to [1]_
+                :func:`fwdti.nls_fit_tensor`
+
+            callable has to have the signature:
+              fit_method(design_matrix, data, *args, **kwargs)
+        args, kwargs : arguments and key-word arguments passed to the
+           fit_method. See fwdti.wls_fit_tensor, fwdti.nls_fit_tensor for
+           details
+        min_signal : float
+            The minimum signal value. Needs to be a strictly positive
+            number. Default: minimal signal in the data provided to `fit`.
+
+        References
+        ----------
+        .. [1] Hoy, A.R., Koay, C.G., Kecskemeti, S.R., Alexander, A.L., 2014.
+               Optimization of a free water elimination two-compartmental model
+               for diffusion tensor imaging. NeuroImage 103, 323-333.
+               doi: 10.1016/j.neuroimage.2014.09.053
+        """
+        ReconstModel.__init__(self, gtab)
```

Comment:
Strangely, `coverage` is reporting that the lines 113-121 are not being tested. That must be an error with `coverage` right?